File: Move Zeroes.c
void moveZeroes(int* nums, int numsSize)
{
    int buf, i, flag = 1;

    while (flag == 1)
    {
        i = 0;
        flag = 0;
        while (i < numsSize - 1)
        {
            if (nums[i] == 0 && nums[i+1] != 0)
            {
                buf = nums[i + 1];
                nums[i + 1] = 0;
                nums[i] = buf;
                flag = 1;
            }
            i++;
        }
    }
}

File: MoveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if len(nums) == 0:
            return nums

        non_zero_len = len(nums)

        i = 0
        while i < non_zero_len:
            if nums[i] == 0:
                zero = nums.pop(i)
                nums.append(zero)
                non_zero_len -= 1
                i -= 1
            i += 1

        return nums
        